qiskit/wrapper/configrc/_settings.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Module that holds the runtime location of qiskitrc file.
+"""
+
+QISKIT_RC_FILE = None
+
+REGISTER_CALLED = 0
```

Comment:
Same here, this is related to [`_register.py`](https://github.com/QISKit/qiskit-core/pull/547/files#diff-3f63a036583755b6e45daf9676874e93R1) only and so, it should live there.

Reply:
QISKIT_RC_FILE is used in several places.  The other one can be moved.

Reply:
Actually REGISTER_CALLED is also used in three different modules.

Reply:
According to my search results of your branch, the name `REGISTER_CALLED` appears indeed in two different places (in addition to this file): `_register.py` and `_wrapper.py` whereas `QISKIT_RC_FILE` appears only in `_settings.py`. So, you can move `QISKIT_RC_FILE` there.

The problem with `REGISTER_CALLED` arises from the need for communicating between `_wrapper.py` and `_register.py`. This is one solution:

1. Move [the registration code in `_wrapper.py`](https://github.com/nonhermitian/qiskit-core/blob/206403188db9508a5bc6adbd159a9337550b63ea/qiskit/wrapper/_wrapper.py#L45-L71) into another function `_register_using_origin_chain`. And make `register` function to delegate into this new function.
2. Move this function inside `_register.py` and adjust the imports in `_wrapper.py`.

Now all `REGISTER_CALLED` only appears in `_register.py` and you can turn the global variable into a module-level one inside `_register.py`.

Reply:
I can do this.  It makes the old `register` function just a wrapper for one inside of `_register` but I am fine with that.